client/state/data-layer/wpcom/sites/automated-transfer/eligibility/index.js

```diff
@@ -111,41 +111,33 @@ const trackEligibility = data => {
 /**
  * Issues an API request to fetch eligibility information for a site
  *
- * @param {Object} store Redux store
  * @param {Function} store.dispatch action dispatcher
  * @param {action} action Action object
+ * @return {Object} action
  */
-export const requestAutomatedTransferEligibility = ( { dispatch }, action ) => {
-	const { siteId } = action;
-
-	dispatch(
-		http(
-			{
-				method: 'GET',
-				path: `/sites/${ siteId }/automated-transfers/eligibility`,
-				apiVersion: '1',
-			},
-			action
-		)
+export const requestAutomatedTransferEligibility = action =>
+	http(
+		{
+			method: 'GET',
+			path: `/sites/${ action.siteId }/automated-transfers/eligibility`,
+			apiVersion: '1',
+		},
+		action
 	);
-};
 
-export const updateAutomatedTransferEligibility = ( { dispatch }, { siteId }, data ) => {
-	dispatch(
-		withAnalytics( trackEligibility( data ), updateEligibility( siteId, fromApi( data ) ) )
-	);
-};
+export const updateAutomatedTransferEligibility = ( { siteId }, data ) =>
+	withAnalytics( trackEligibility( data ), updateEligibility( siteId, fromApi( data ) ) );
 
-export const throwRequestError = ( store, action, error ) => {
+export const throwRequestError = ( action, error ) => {
 	throw new Error( error );
```

Comment:
I know it's pre-existing code but... Throwing exceptions in these handlers doesn't make sense. The action dispatch will throw and nobody is catching the exception. In the end, after you remove all the abstraction layers, it's equivalent to code like this:
```js
wpcom.req.get( eligibilityEndpoint, ( error, data ) => {
  if ( error ) {
    throw new Error( error );
  }
} );
```

Reply:
Yes, `noop` seems to be the best approach here. Certainly better that throwing an exception that won't be handled.